tests/tx/test_tx.py

```diff
@@ -111,6 +113,34 @@ def test_checkpoint_validation(self):
         self.assertTrue(self.manager.on_new_tx(block2, sync_checkpoints=True, partial=True, fails_silently=False))
         self.assertEqual(block2.get_metadata().validation, ValidationState.CHECKPOINT_FULL)
 
+        # initialize a new manager and try to add a checkpoint block before its parents
+        block2_height = block2.get_metadata().height
+        manager2 = self.create_peer('testnet', checkpoints=[Checkpoint(block2_height, block2.hash)])
+        del block2._metadata
+        block2.storage = manager2.tx_storage
+        block2.set_height(block2_height)
+        # it should fail if we don't pass sync_checkpoints=True
+        with self.assertRaises(InvalidNewTransaction):
+            manager2.on_new_tx(block2, partial=True, fails_silently=False)
+        # also test if it fails silently
+        self.assertFalse(manager2.on_new_tx(block2, partial=True, fails_silently=True))
+        # otherwise it should be accepted
+        self.assertTrue(manager2.on_new_tx(block2, sync_checkpoints=True, partial=True, fails_silently=False))
+        self.assertEqual(block2.get_metadata().validation, ValidationState.CHECKPOINT)
+
+        # XXX: this stopped failing because we removed the partial children metadata which simplified the validation
```

Comment:
Maybe we should remove it from here and move to another branch.

Reply:
Yes, I'll do that.